test/rekt/resources/apiserversource/apiserversource_test.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserversource_test
+
+import (
+	"os"
+
+	v1 "knative.dev/eventing/pkg/apis/sources/v1"
+
+	"knative.dev/eventing/test/rekt/resources/apiserversource"
+
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/reconciler-test/pkg/manifest"
+)
+
+// The following examples validate the processing of the With* helper methods
+// applied to config and go template parser.
+
+func Example_min() {
+	images := map[string]string{}
+	cfg := map[string]interface{}{
+		"name":      "foo",
+		"namespace": "bar",
+	}
+
+	files, err := manifest.ExecuteLocalYAML(images, cfg)
+	if err != nil {
+		panic(err)
+	}
+
+	manifest.OutputYAML(os.Stdout, files)
+	// Output:
+	// apiVersion: sources.knative.dev/v1
+	// kind: ApiServerSource
+	// metadata:
+	//   name: foo
+	//   namespace: bar
+	// spec:
+}
+
+func Example_withServiceAccountName() {
+	images := map[string]string{}
+	cfg := map[string]interface{}{
+		"name":      "foo",
+		"namespace": "bar",
+	}
+
+	apiserversource.WithServiceAccountName("src-sa")(cfg)
+
+	files, err := manifest.ExecuteLocalYAML(images, cfg)
+	if err != nil {
+		panic(err)
+	}
+
+	manifest.OutputYAML(os.Stdout, files)
+	// Output:
+	// apiVersion: sources.knative.dev/v1
+	// kind: ApiServerSource
+	// metadata:
+	//   name: foo
+	//   namespace: bar
+	// spec:
+	//   serviceAccountName: src-sa
+}
+
+func Example_withEventMode() {
+	images := map[string]string{}
+	cfg := map[string]interface{}{
+		"name":      "foo",
+		"namespace": "bar",
+	}
+
+	apiserversource.WithEventMode(v1.ReferenceMode)(cfg)
+
+	files, err := manifest.ExecuteLocalYAML(images, cfg)
+	if err != nil {
+		panic(err)
+	}
+
+	manifest.OutputYAML(os.Stdout, files)
+	// Output:
+	// apiVersion: sources.knative.dev/v1
+	// kind: ApiServerSource
+	// metadata:
+	//   name: foo
+	//   namespace: bar
+	// spec:
+	//   mode: Reference
+}
+
+func Example_withSink() {
+	images := map[string]string{}
+	cfg := map[string]interface{}{
+		"name":      "foo",
+		"namespace": "bar",
+	}
+
+	sinkRef := &duckv1.KReference{
+		Kind:       "sinkkind",
+		Namespace:  "sinknamespace",
```

Comment:
I wonder if above this we might want to make the comment about namespace being not used? Or above where you skipped it? Cause here when you create it with this namespace, it might not be clear to the user why L136 is not sinknamespace?

Reply:
I pushed a commit for not skipping the namespace.